src/arrayMethodReduce.js

```diff
@@ -5,7 +5,21 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let acc;
+    let loopStart = 0;
+
+    if (initialValue !== undefined) {
+      acc = initialValue;
+    } else {
+      acc = this[0];
+      loopStart++;
```

Comment:
there's no need to use incrementation here. You need only either 0 or 1. Nothing more